src/inc/msquic.h

```diff
@@ -1605,6 +1605,56 @@ QUIC_STATUS
     _In_opt_ void* ClientSendContext
     );
 
+//
+// Connection Pool API
+//
+
+#ifdef QUIC_API_ENABLE_PREVIEW_FEATURES
+
+typedef enum QUIC_CONNECTION_POOL_FLAGS {
+    QUIC_CONNECTION_POOL_FLAG_NONE =                            0x00000000,
+    QUIC_CONNECTION_POOL_FLAG_CLOSE_CONNECTIONS_ON_FAILURE =    0x00000001,
+} QUIC_CONNECTION_POOL_FLAGS;
+
+DEFINE_ENUM_FLAG_OPERATORS(QUIC_CONNECTION_POOL_FLAGS);
+
+typedef struct QUIC_CONNECTION_POOL_CONFIG {
+    _In_ HQUIC Registration;
+    _In_ HQUIC Configuration;
+    _In_ QUIC_CONNECTION_CALLBACK_HANDLER Handler;
+    _Field_size_opt_(NumberOfConnections)
+        void** Context;                         // Optional
+    _In_ const char* ServerName;
+    _In_opt_ const QUIC_ADDR* ServerAddress;    // Optional
```

Comment:
I am still not sure I understand why the server address is provided there, instead of relying only on the server name like in the existing connection start API (with an ip address being potentially passed in text form as the server name).

IMO, we should be consistent in the API pattern.

Reply:
So the ServerName doubles as the SNI name, which is used in TLS. You *must* specify the ServerName, because TLS 1.3 requires SNI (at least our implementation does).
In some deployments, you can have DNS-based load balancing, where your DNS name could resolve to multiple IP addresses. A caller may want to control which server they connect to specifically. Our solution to that is to allow the caller to specify the SNI as the ServerName, but then set the RemoteAddress via the SetParam interface before calling Start.
Because this API creates the connection *and* starts it, leaving the caller no opportunity to call the SetParam interface, we must expose that via this API.

Reply:
Thanks for the details.
It would probably be worth adding this to the documentation, to clarify when the `ServerAddress` is expected and what is a valid value (I assume something will fail if the DNS resolution doesn't return the specific address in its list?)